### DISCUSSION TITLE
How to deal with UML first approach ?

### DISCUSSION THREAD

**Discussion by fliot** [sqlalchemy/sqlalchemy#9813] [Ideas]
Hi,

Starting a full new application, I would love to start by an UML Class diagram, and create SQL and SQLAlchemy classes automatically.
Do you know some UML designer that support SQLAlchemy classes generation ?

I saw it's possible from an SQL database to generate SQLAlchemy classes, but from what I tested, it's not fabulous to deal with foreign keys constrains (and I'm even not speaking about 1..n cardinality, which would also matter at some point).

Any advice for UML first approach until SQLAlchemy ?
I would love to hear your thoughts.

Thanks in advance.

**Comment by zzzeek** [comment]
SQLAlchemy models tend to derive from database schemas more than UML.    The https://github.com/agronholm/sqlacodegen tool can generate models from an existing schema.   To design database schemas graphically you would use an ER design tool, there's lots of those available online and such, free, commercial etc.
